helix-core/src/test/java/org/apache/helix/integration/task/TestForceDeleteWorkflow.java

```diff
@@ -0,0 +1,368 @@
+
+    // Wait until workflow is stopped.
+    _driver.pollForWorkflowState(workflowName, TaskState.STOPPED);
+
+    // Wait until workflow is stopped. Also, jobs should be either stopped or not created (null).
+    boolean isWorkflowStopped = TestHelper.verify(() -> {
+      WorkflowContext wCtx1 = _driver.getWorkflowContext(workflowName);
+      TaskState job0 = wCtx1.getJobState(TaskUtil.getNamespacedJobName(workflowName, "JOB0"));
+      TaskState job1 = wCtx1.getJobState(TaskUtil.getNamespacedJobName(workflowName, "JOB1"));
+      TaskState job2 = wCtx1.getJobState(TaskUtil.getNamespacedJobName(workflowName, "JOB2"));
+      return ((job0 == null || job0 == TaskState.STOPPED)
+          && (job1 == null || job1 == TaskState.STOPPED)
+          && (job2 == null || job2 == TaskState.STOPPED));
+    }, 60 * 1000);
+    Assert.assertTrue(isWorkflowStopped);
+
+    // Check that WorkflowConfig, WorkflowContext, and IdealState are indeed created for this
+    // workflow.
+    Assert.assertNotNull(_driver.getWorkflowConfig(workflowName));
+    Assert.assertNotNull(_driver.getWorkflowContext(workflowName));
+    Assert.assertNotNull(_admin.getResourceIdealState(CLUSTER_NAME, workflowName));
+
+    // Stop the Controller
+    _controller.syncStop();
+
+    _driver.delete(workflowName, true);
+
+    // Check that WorkflowConfig, WorkflowContext, and IdealState are indeed deleted for this
+    // workflow.
+    boolean isWorkflowDeleted = TestHelper.verify(() -> {
+      WorkflowConfig wfcfg = _driver.getWorkflowConfig(workflowName);
+      WorkflowContext wfctx = _driver.getWorkflowContext(workflowName);
+      IdealState is = _admin.getResourceIdealState(CLUSTER_NAME, workflowName);
+      return (wfcfg == null && wfctx == null && is == null);
+    }, 60 * 1000);
+    Assert.assertTrue(isWorkflowDeleted);
+  }
+
+  @Test(dependsOnMethods = "testDeleteStoppedWorkflowForcefully")
+  public void testDeleteStoppingStuckWorkflowForcefully() throws Exception {
+    // In this test, Stuck workflow indicates a workflow that is in STOPPING state (user requested
+    // to stop the workflow), its JOBs are also in STOPPING state but the tasks are stuck (or taking
+    // a long time to stop) in RUNNING state.
+
+    // Reset the cancel and stop counts
+    CANCEL_COUNT.set(0);
+    STOP_COUNT.set(0);
+
+    // Start the Controller
+    String controllerName = CONTROLLER_PREFIX + "_0";
+    _controller = new ClusterControllerManager(ZK_ADDR, CLUSTER_NAME, controllerName);
+    _controller.syncStart();
+
+    String workflowName = TestHelper.getTestMethodName();
+    Workflow.Builder builder = createCustomWorkflow(workflowName, LONG_EXECUTION_TIME, STOP_DELAY);
+    _driver.start(builder.build());
+
+    // Wait until workflow is created and running.
+    _driver.pollForWorkflowState(workflowName, TaskState.IN_PROGRESS);
+
+    _driver.stop(workflowName);
+    // Wait until workflow is stopped.
+
+    // Wait until workflow is created and running.
```

Comment:
The comment and the code don't match. If the workflow is created and running, then the workflow state should be in STOPPING? That doesn't make any sense?